tests/php/test-tag-and-attribute-sanitizer.php

```diff
@@ -2463,8 +2465,32 @@ public function get_html_data() {
 				[],
 				[ AMP_Tag_And_Attribute_Sanitizer::INVALID_CDATA_HTML_COMMENTS ],
 			],
+			'cdata_malformed_json'                    => [
+				'<html><head><meta charset="utf-8"><script type="application/ld+json">{"example": </script></head><body></body></html>',
+				'<html><head><meta charset="utf-8"></head><body></body></html>',
+				[],
+				[ AMP_Tag_And_Attribute_Sanitizer::INVALID_JSON_CDATA ],
+			],
+			'cdata_malformed_json_with_emojis'        => [
+				'<html><head><meta charset="utf-8"><script type="application/ld+json">{"wrong": 🚧 🚧 }</script></head><body></body></html>',
```

Comment:
This test isn't quite capturing the original issue observed. This should do it:

```suggestion
				'<html><head><meta charset="utf-8"><script type="application/ld+json">{"wrong": "<?php echo wp_staticize_emoji( "🚧 🚧" ); ?>"}</script></head><body></body></html>',
```

Reply:
OK, committed in https://github.com/ampproject/amp-wp/pull/4340/commits/5b187d36165ee6361d494039b78a7772204876b4